fiatconnect-api.md

```diff
@@ -391,105 +407,204 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
 
-The rationale behind not requiring an `exp` claim for these endpoints is that these endpoints are *non-mutating*; calling them does not affect server-side
-state in any way. Since these endpoints are less security-critical, an `exp` claim is not required as a convenience to clients.
-While these endpoints do not require the `exp` claim to be present, *all endpoints* MUST honor the `exp` claim *if it is* present in the
-JWT payload. For endpoints where the `exp` claim is required, the server MUST return an HTTP `400` if it is not present.
-A server MUST return an HTTP `401` error if the provided `exp` field is later than the current time with respect to Epoch. If an `exp` claim is
-included in a request, it MUST NOT exceed 24 hours from the current time with respect to Epoch. If the `exp` claim is more than 24 hours into the future, the
-server MUST return an HTTP `400` error. This is required in order to prevent clients from generating excessivley long-lived JWTs that would pose a security
-risk if leaked.
+### 3.2.2. Client API Key
 
-##### 3.2.1.2.4. `"client"` Claim
+In addition to the SIWE-based auth used to authenticate users and create sessions, the FiatConnect specification also supports a more traditional authentication standard, used
+to identify specific clients with requests. Recall the webhook-based status monitoring mentioned earlier in this document. In order to support status
+monitoring via webhooks, individual clients will need to be able to register a URL pointing to an API able to handle webhook updates from the server.
+Once a client has registered a webhook URL with the provider, the client needs a way to identify itself to the server. To uniquely identify clients in order
+to know where to send webhook-based status updates, a server may allow clients to register an API key. The exact mechanism by which servers allocate API keys
+to clients and allow them to register webhook URLs is out of scope of this document.
 
-In addition to public-private key authentication, servers must also support API token based authentication. Recall the webhook-based status monitoring
-mentioned earlier in this document. In order to support status monitoring via webhooks, individual clients will need to be able to register a URL pointing
-to an API able to handle webhook updates from the server. Once a client has registered a webhook URL with the provider, the client needs a way to identify
-itself to the server. To uniquely identify clients in order to know where to send webhook-based status updates, a server may allow clients to register an API key.
-The exact mechanism by which servers allocate API keys to clients and allow them to register webhook URLs is out of scope of this document.
+Once a client has registered an API key with a provider and associated it with a webhook URL, a client MAY provide it to a server through an `Authorization` header,
```

Comment:
It's unclear to me how the session cookie and API key interact? If a user provides an API key, but no session cookie, does that authenticate them?

Reply:
No, they're totally separate. The API token is merely used to identify a request with a particular client (specifically, for the purpose of knowing where to send webhook events). I've clarified this in the doc a bit.